### PR TITLE
refactor(ci): redo netsim CI

### PR DESCRIPTION
**Pull request by Arqu**
## Description

Refactors the CI jobs for netsim. Groups it to a single parametrized runner job which is also now a callable workflow. 
Makes use of the new parallel runners and also has a quick job for just testing perf.

## Breaking Changes

Removing the `/netsim` comment commands in favor of running it from the actions menu.

## Change checklist

- [ ] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (flub) [inline .github/workflows/ci.yml:1]: love this splitting off into its own file

File: .github/workflows/netsim.yml [hunk 1]
```diff
@@ -4,8 +4,22 @@ on:
...
+      pr_number:
+        required: true
+        type: string
+      branch:
+        required: true
+        type: string
+      netsim_branch:
+          required: true
+          type: string
+          default: "main"
```
Comment (flub) [inline .github/workflows/netsim.yml:18]: nit: inconsistent indentation

File: .github/workflows/netsim_runner.yaml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        ./setup.sh
+        ./cleanup.sh || true
+
+    - name: Copy binaries to right location
+      run: |
+        cp target/${{inputs.build_profile}}/iroh ../chuck/netsim/bins/iroh
+        cp target/${{inputs.build_profile}}/iroh-relay ../chuck/netsim/bins/iroh-relay
+        cp ../chuck/target/release/chuck ../chuck/netsim/bins/chuck
+
+    - name: Get commit sha
```
Comment (flub) [inline .github/workflows/netsim_runner.yaml:138]: kind of surprised github doesn't provide this already in some kind of variable

File: .github/workflows/netsim_runner.yaml [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        export AWS_DEFAULT_REGION=us-west-2
+        
+        curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+        unzip -q awscliv2.zip
+        sudo ./aws/install --update
+
+        cd ../chuck/netsim
+
+        aws_fname=${{ env.LAST_COMMIT_SHA }}.tar.gz
+        tar cvzf report.tar.gz report_prom.txt report_table.txt report_metro.txt report_metro_integration.txt logs/ report/ viz/
```
Comment (flub) [inline .github/workflows/netsim_runner.yaml:174]:
```suggestion
        tar -cvzf report.tar.gz report_prom.txt report_table.txt report_metro.txt report_metro_integration.txt logs/ report/ viz/
```
doesn't really matter, but we have modern tools these days.  no need to pretend they're stuck in the 80s ;)

**Review by flub — APPROVED** [review]

**Comment by flub** [comment]
> Removing the `/netsim` comment commands in favor of running it from the actions menu.

good choice

File: .github/workflows/netsim_runner.yaml (hunk 2 above)
Comment (Arqu) [reply .github/workflows/netsim_runner.yaml:138]: Kinda does, but also kinda doesn't. Depending on how the job is called it uses a completely different set of `github.*.*` variable so it's pretty janky. This is one surefire way to get the commit hash you're currently checked out at.

**Pull request merged**